src/test/java/net/dv8tion/jda/test/modals/ModalTest.java

```diff
@@ -0,0 +1,96 @@
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.test.modals;
+
+import net.dv8tion.jda.api.components.Component;
+import net.dv8tion.jda.api.components.attribute.IDisableable;
+import net.dv8tion.jda.api.components.label.Label;
+import net.dv8tion.jda.api.components.label.LabelChildComponent;
+import net.dv8tion.jda.api.components.selections.EntitySelectMenu;
+import net.dv8tion.jda.api.components.selections.StringSelectMenu;
+import net.dv8tion.jda.api.modals.Modal;
+import net.dv8tion.jda.test.components.ComponentTestData;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Stream;
+
+public class ModalTest {
+    static Map<Component.Type, Class<? extends LabelChildComponent>> LABEL_CHILDREN = Map.of(
+            Component.Type.STRING_SELECT, StringSelectMenu.class,
+            Component.Type.USER_SELECT, EntitySelectMenu.class,
+            Component.Type.ROLE_SELECT, EntitySelectMenu.class,
+            Component.Type.MENTIONABLE_SELECT, EntitySelectMenu.class,
+            Component.Type.CHANNEL_SELECT, EntitySelectMenu.class);
+
+    @MethodSource("modalComponents")
+    @ParameterizedTest
+    public void testDisabledComponents(Component component) {
+        Assertions.assertInstanceOf(IDisableable.class, component);
+        Assertions.assertInstanceOf(LabelChildComponent.class, component);
+        LabelChildComponent mutatedComponent = ((LabelChildComponent) ((IDisableable) component).asDisabled());
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            Modal.create("id", "title")
+                    .addComponents(Label.of("label", mutatedComponent))
+                    .build();
+        });
+    }
+
+    @MethodSource("modalComponents")
+    @ParameterizedTest
+    public void testEnabledComponents(Component component) {
+        // Enabled is the default state
+        Assertions.assertInstanceOf(LabelChildComponent.class, component);
+        Assertions.assertDoesNotThrow(() -> {
+            Modal.create("id", "title")
+                    .addComponents(Label.of("label", (LabelChildComponent) component))
+                    .build();
+        });
+    }
+
+    @Test
+    public void testOneInvalidComponentInList() {
+        var components = new ArrayList<>(testComponents().toList());
+        var mutatedComponentIndex = ThreadLocalRandom.current().nextInt(components.size());
+        Assertions.assertInstanceOf(IDisableable.class, components.get(mutatedComponentIndex));
+        components.set(mutatedComponentIndex, ((IDisableable) components.get(mutatedComponentIndex)).asDisabled());
+
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
```

Comment:
If you extend `AbstractSnapshotTest`, you can snapshot the exception output:

```java
assertThatIllegalArgumentException()
    .isThrownBy(() -> {
        ...
    })
    .satisfies(exception -> assertWithSnapshot(exception.toString(), null));
```

Feel free to add `assertWithSnapshot(String)` overload to avoid the `null` parameter. To update snapshots you can either run the test with intellij's junit test runner or using the gradle task `./gradlew updateTestSnapshots`